substrate/frame/revive/src/benchmarking/code.rs

```diff
@@ -95,9 +95,22 @@ impl WasmModule {
 		Self::new(bench_fixtures::NOOP.to_vec())
 	}
 
-	/// A contract code that executes some ALU instructions in a loop.
+	/// A contract code that does unaligned memory accessed in a loop.
 	pub fn instr() -> Self {
-		Self::new(bench_fixtures::INSTR.to_vec())
+		let text = "
+		pub @deploy:
+		ret
+		pub @call:
+			@loop:
+				jump @done if t0 == a1
+				a0 = u64 [a0]
+				t0 = t0 + 1
+				jump @loop
+			@done:
```

Comment:
We are benchmarking one loop count per `r`. So we are overestimating here. by measuring 4 instructions instead of one. This is on top of the already existing overestimation: Most of a contracts instructions will not be cache misses let alone unaligned reads.

We **could** unroll the loop I guess. But then we will be measuring the lazy compilation of those max sized basic blocks, too.

We should probably subtract something from the final `Weight`. As of right now we are dividing it by 6 because this is what the old code did. But it has to be reworked. Should we take the value on face value or subtract or divide it? cc @koute

Reply:
If you want to get rid of the loop overhead you could run this measurement twice: once with `a0 = u64[a0]` inside, and then again *without*, and you have a baseline which you can use to get rid of the loop overhead.

Reply:
Good idea. Will try that.